src/main/kotlin/com/hibob/oop/MeetingModel.kt

```diff
@@ -0,0 +1,39 @@
+package com.hibob.oop
+
+data class Participant(
+    val name : String,
+    val email : String
+)
+
+open class Meeting(
+    open val name : String,
+    open val location : Location,
+    private val participants : List<Participant>
+) {
+
+    open fun addParticipant(participant: Participant) : Meeting =
+        Meeting(name, location, participants + participant)
```

Comment:
since the list of participants is unmutable if you won'r return it the list won't change from the outside. so either change the list to mutable or return the new object